[docs] Voyager docs for v1.3 (#17505)
yugabyte/yugabyte-db: docs/content/preview/migrate/performance.md

- **Add disks** to reduce disk write contention. YugabyteDB servers can be configured with one or multiple disk volumes to store tablet data. If all tablets are writing to a single disk, write contention can slow down the ingestion speed. Configuring the [YB-TServers](../../reference/configuration/yb-tserver/) with multiple disks can reduce disk write contention, thereby increasing throughput. Disks with higher IOPS and better throughput also improve write performance.

> - **Enable packed rows** (Beta) to increase the throughput by more than two times. Enable packed rows on the YugabyteDB cluster by setting the YB-TServer GFlag `ysql_enable_packed_row` to true.

Reviewer: early access +link

Reviewer 2: @aishwarya24, yes please link `Enable packed rows (Beta)` with - https://docs.yugabyte.com/preview/reference/configuration/yb-tserver/#packed-row-flags-beta

> - **Enable packed rows** (Beta) to increase the throughput by more than two times. Enable packed rows on the YugabyteDB cluster by setting the YB-TServer GFlag `ysql_enable_packed_row` to true.

Reviewer: ```suggestion
- **Enable packed rows** (Early Access) to increase the throughput by more than two times. Enable packed rows on the YugabyteDB cluster by setting the YB-TServer flag `ysql_enable_packed_row` to true.
```

- **Configure the host machine's disk** with higher IOPS and better throughput to improve the performance of the splitter, which splits the large data file into smaller splits of 20000 rows. Splitter performance depends on the host machine's disk.

> **Test results**

Reviewer 3: `Performance Test Results`

> **Performance test results**

Reviewer: why not an h4?

> Following are the performance test results to demonstrate the preceding techniques:

Reviewer: ```suggestion
The following performance test results demonstrate the preceding techniques.
```

yb-voyager was tested using varying configurations, including more parallel jobs, multiple disks, and a larger cluster. The tests were run using a 28GB CSV file with 350 million rows on YugabyteDB version 2.16.0.0-b90.

> 

Reviewer 2: `import data` command was tested using varying configurations, including more parallel jobs, ....

By default, yb-voyager exports four tables at a time. To improve data export, parallelize the export of data from multiple tables using the [–-parallel-jobs](../reference/yb-voyager-cli/#parallel-jobs) argument with the export data command to increase the number of jobs. Setting the value too high can however negatively impact performance; a setting of '4' typically performs well.

> If you use EXPERIMENTAL_FAST_EXPORT to [accelerate data export](../migrate-steps/#accelerate-data-export-for-mysql-or-oracle), `--parallel-jobs` will not have any impact on performance and yb-voyager will export only one table at a time.

Reviewer 4: **Note to the developers:**
@sanyamsinghal @makalaaneesh shall we log a WARNING also in the code for this - if this arg is used and the accelerated export is attempted then just show a warning on the screen.

Reviewer 5: yeah i think it can be helpful also , in case user is not aware of the env var is set and assuming normal export.

> ## Test results

Reviewer 3: @aishwarya24 A small restructure required here: 

Please move the `Test Results` section to under [Improve import performance > Techniques to improve performance](https://deploy-preview-17505--infallible-bardeen-164bc9.netlify.app/preview/migrate/performance/#techniques-to-improve-performance) (just above the note) introduction: `Performance test results to demonstrate the above techniques:`

{{< note title="Note" >}}

> - With increased optimisations, the CPU usage also shoots up. Permitting the CPU usage beyond 60% may not be advisable as the database requires some cycles to do it's internal operations.

Reviewer: ```suggestion
- The more optimizations you use, the more CPU you will require. Allowing CPU use greater than 60% may not be advisable as the database requires some cycles for internal operations.
```

- These performance optimizations apply whether you are importing data using the yb-voyager [import data command](../migrate-steps/#import-data) or the [import data file command](../migrate-steps/#import-data-file).

{{< /note >}}

## Improve export performance

> By default, yb-voyager exports four tables at a time. To improve data export, parallelize the export of data from multiple tables using the [–-parallel-jobs](../reference/yb-voyager-cli/#parallel-jobs) argument with the export data command to increase the number of jobs. Setting the value too high can however negatively impact performance; a setting of '4' typically performs well.

Reviewer: ```suggestion
By default, yb-voyager exports four tables at a time. To speed up data export, parallelize the export of data from multiple tables using the [–-parallel-jobs](../reference/yb-voyager-cli/#parallel-jobs) argument with the export data command to increase the number of jobs. Setting the value too high can, however, negatively impact performance; a setting of 4 typically performs well.
```

> By default, yb-voyager exports four tables at a time. To improve data export, parallelize the export of data from multiple tables using the [–-parallel-jobs](../reference/yb-voyager-cli/#parallel-jobs) argument with the export data command to increase the number of jobs. Setting the value too high can however negatively impact performance; a setting of '4' typically performs well.

Reviewer: Not clear if yb-voyager is setting parallel-jobs to 4 by default, or what. number of jobs = number of tables exported in parallel?

Author: clarified the defaults.

> If you use BETA_FAST_DATA_EXPORT to [accelerate data export](../migrate-steps/#accelerate-data-export-for-mysql-and-oracle), `--parallel-jobs` will not have any impact on performance and yb-voyager will export only one table at a time.

Reviewer: ```suggestion
If you use BETA_FAST_DATA_EXPORT to [accelerate data export](../migrate-steps/#accelerate-data-export-for-mysql-and-oracle), yb-voyager exports only one table at a time and the `--parallel-jobs` argument is ignored.
```
